### PR TITLE
update forking and saving VM state and misc on run.md

### PR DESCRIPTION
**Pull request by ryestew**
Pinning, saving state, and general update to deploy and run.

### CODE REVIEW

**Review by vermouth22 — CHANGES_REQUESTED** [review]
I made a brand PR as an edit: https://github.com/ethereum/remix-ide/pull/3202

File: docs/run.md [hunk 1]
```diff
@@ -3,48 +3,87 @@ Deploy & Run
...
+For assistance with getting the configuration (the RPC server address etc), click the plug icon next the word **ENVIRONMENT** to get to chainlist.org. 
+
+![](images/a-deploy-run-plug.png)
+
+
+Remix VM
 ---------
+The Remix VM is a sandbox blockchain in the browser.  Transactions do not require an approval to run.  It comes with 10 accounts; each loaded with 100ETH.
+
+In the current version of Remix, the **state** of the chain is saved in the **.states folder** in File Explorer.  This was not the case in earlier versions of Remix - where this chain would refresh when the browser would refresh.
```
Comment (Aniket-Engg) [inline docs/run.md:32]: chain would ~~refresh~~ `reset` when the ~~browser~~ Remix would refresh.

File: docs/run.md [hunk 2]
```diff
@@ -3,48 +3,87 @@ Deploy & Run
...
+
+To turn off the saving of the Remix VM’s state, uncheck the **Save environment state** in the Settings panel.
+
+Saving the state means you can refresh the browser and not lose your work.  The caviot being that storage in a browser is inherently feable.  Of course, if you push to a remote repo or if you use Remixd to share a folder on your hard drive, then you have are not relying on the browser to save your work.
```
Comment (Aniket-Engg) [inline docs/run.md:36]: refresh the ~~browser~~ Remix IDE

as Browser can have many tabs

Comment (Aniket-Engg) [reply docs/run.md:36]: ~~caviot~~ caveat

File: docs/udapp.md [hunk 3]
```diff
@@ -1,19 +1,26 @@
...
 
-The deployed contract appears but is in its collapsed form.
+![](images/a-deploy-run-deployed-instances.png)
 
-![](images/a-debug2-udapp1a.png)
+The deployed contract's address is visible as are a few other icons - one of which is the **pin** icon.  
 
-Click the sideways caret to open it up.
+## Pinned contracts
+When a contract is pinned, it will jump up to the **Pinned Contracts** section and Remix will save the contract's address and the ABI (in the .deploys folder of the File Explorer).  When the browser is refreshed, the pinned contracts will be loaded into the Pinned Contracts section.
```
Comment (Aniket-Engg) [inline docs/udapp.md:13]: .deploys folder of the ~File Explorer~ `current workspace`

Comment (Aniket-Engg) [reply docs/udapp.md:13]: When the ~~browser~~ Remix is refreshed

File: docs/udapp.md [hunk 4]
```diff
@@ -1,19 +1,26 @@
...
+#### Pinned contracts are chain & Workspace specific
+Because a pinned contract's address and ABI are stored in a File Explorer Workspace, the same Workspace must be active to see its pinned contracts.  Similarly, only the pinned contracts of the currently selected chain will show.
 
-You will see the functions in the contract.  The functions buttons can have different color buttons.
+## Functions 
+TO see a contract's functions, click the caret on the left side of the panel.
+
+![](images/a-deploy-run-caret.png)
+
+The functions buttons can have different color buttons.
```
Comment (Aniket-Engg) [inline docs/udapp.md:23]: have different color ~buttons~

**Review by Aniket-Engg — CHANGES_REQUESTED** [review]

File: docs/udapp.md [hunk 5]
```diff
@@ -20,13 +20,13 @@ TO see a contract's functions, click the caret on the left side of the panel.
 
 ![](images/a-deploy-run-caret.png)
 
-The functions buttons can have different color buttons.
+The functions' buttons can have different colors.
 
--   Functions that are `constant` or `pure` functions in Solidity have a blue buttons. Clicking one of this type does not create a new transaction. So clicking will not cause state changes - it will only return a value stored in the contract - so it won't cost you anything in gas fees.
+-   Blue buttons are for `constant` or `pure` functions. Clicking a blue button does not create a new transaction - so there will be **no gas fees**.
```
Comment (Aniket-Engg) [inline docs/udapp.md:25]: Blue buttons are for ~constant~ `view` or `pure`

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**